diff --git a/public_html/index.php b/public_html/index.php
index 79c632b..6a3c75b 100644
--- a/public_html/index.php
+++ b/public_html/index.php
@@ -212,6 +212,10 @@ elseif ($RCMAIL->task != 'login' && !empty($_SESSION['user_id'])) {
     if (!$RCMAIL->session->check_auth()) {
         $session_error = $RCMAIL->session_error() ?: 'sessionerror';
         $RCMAIL->kill_session();
+        if (!empty($_SERVER['PHP_AUTH_USER'])) {
+            header('Location: https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI']);
+            exit;
+        }
     }
 }
 
